packages/compass-aggregations/src/modules/pipeline-builder/stage-editor.ts

```diff
@@ -417,21 +417,19 @@ const ESCAPED_STAGE_OPERATORS = STAGE_OPERATORS.map((stage) => {
 
 function getStageSnippet(
   stageOperator: string | null,
-  env: string,
+  env: ServerEnvironment,
   shouldAddComment: boolean,
   escaped = false
 ) {
-  const stage = (escaped ? ESCAPED_STAGE_OPERATORS : STAGE_OPERATORS).find(
-    (stageOp) => {
-      return (
-        stageOp.value === stageOperator &&
-        (stageOp.env as readonly typeof ENVS[number][]).includes(
-          env as typeof ENVS[number]
-        )
-      );
-    }
+  const stages = (escaped ? ESCAPED_STAGE_OPERATORS : STAGE_OPERATORS).filter(
+    (stageOp) => stageOp.value === stageOperator
   );
 
+  const stage =
```

Comment:
NIT: I think you can use `reduce`, instead of `filter` and `find`. You avoid an additional iteration and IMO it's a bit more explicit on what you want to achieve.

```js
const environments = [ 'a', 'b' ];
const stages = [ 
    { value: '$search', env: [ 'a' ] },
    { value: '$search', env: [ 'a', 'b' ] },
    { value: '$searchMeta', env: [ 'a' ] },
    { value: '$out', env: [ 'a', 'b' ] }
]

function getStageSnippet(stageOperator, env) {
    return stages.reduce((result, stage) => {
        if (stage.value === stageOperator) {
            if (result.env?.includes(env)) {
                return result;
            }
            return stage;
        }

        return result;
    }, {});
}

console.log(getStageSnippet('$search', 'a'))
console.log(getStageSnippet('$search', 'b'))
console.log(getStageSnippet('$searchMeta', 'a'))
console.log(getStageSnippet('$out', 'a'))
console.log(getStageSnippet('$nonexist', 'a'))
```